src/components/cesium_map/api/spatial.js

```diff
@@ -43,6 +43,26 @@ export class SpatialView {
       orientation: this.orientation
     };
   }
+
+  // return position information as list
+  get viewDict() {
+    return {
+      longitude: this.longitude,
+      latitude: this.latitude,
+      height: this.height,
+      heading: this.heading,
+      pitch: this.pitch
+    }
+  }
+
+  // compare if two positions are the same
+  equalTo(position) {
+    return this.longitude.toFixed(8) === position.longitude.toFixed(8) &&
+      this.latitude.toFixed(8) === position.latitude.toFixed(8) &&
+      this.height.toFixed(8) === position.height.toFixed(8) &&
+      this.heading.toFixed(8) === position.heading.toFixed(8) &&
+      this.pitch.toFixed(8) === position.pitch.toFixed(8);
```

Comment:
15 decimal is too precise to compare two positions. I round each attribute to 8 decimal and check if they are equal.